tests/30rooms/13anonymousaccess.pl

```diff
@@ -701,3 +797,18 @@ sub matrix_get_room_membership
       }
    );
 }
+
+sub ignore_presence_for
+{
+   my ( $users, @raw ) = @_;
+
+   my @filtered;
+   foreach my $event (@raw) {
+      my $include = 1;
+      foreach my $user (@$users) {
+         $include = 0 if $event->{type} eq "m.presence" and $event->{content}->{user_id} eq $user->user_id;
+      }
+      push @filtered, $event if $include;
+   }
+   \@filtered
+}
```

Comment:
This is a manually-unrolled way of basically writing a `grep` filter with an `any` test inside. I'd suggest a shorter:

    my ( $ignored_users, @events ) = @_;

    return [ grep {
       my $event = $_;
       not(
          $event->{type} eq "m.presence" and any { $event->{content}{user_id} eq $_->user_id } @$ignored_users
       )
    } @events ];

Or maybe skip the `any` in this case and pull it into a lookup hash instead:

    my %ignored_userids = map { $_->user_id => 1 } @$ignored_users;
    ....
       not ( $event->{type} eq "m.presence" and exists $ignored_userids{ $event->{content}{user_id} } )